category_theory/functor/default.lean

```diff
@@ -0,0 +1,75 @@
+-- Copyright (c) 2017 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Tim Baumann, Stephen Morgan, Scott Morrison
+
+import ..category
+
+open categories
+
+namespace categories.functor
+ 
+universes u₁ v₁ u₂ v₂ u₃ v₃
+
+structure Functor (C : Type u₁) [category.{u₁ v₁} C] (D : Type u₂) [category.{u₂ v₂} D] : Type (max u₁ v₁ u₂ v₂) :=
+  (onObjects     : C → D)
+  (onMorphisms   : Π {X Y : C}, (X ⟶ Y) → ((onObjects X) ⟶ (onObjects Y)))
+  (identities    : ∀ (X : C), onMorphisms (𝟙 X) = 𝟙 (onObjects X) . obviously)
```

Comment:
Now you have `categories.functor.Functor`, I think it should be just `category_theory.functor`.

Reply:
Thanks @johoelzl, I've made all the changes you suggested. I'm very happy to adjust naming to match `mathlib` conventions better, although would like to err on the side of longer names over ambiguous or not-immediately-obvious ones, where reasonable.